lib/daru/vector.rb

```diff
@@ -1481,13 +1481,16 @@ def prepare_bootstrap(estimators)
 
     # Note: To maintain sanity, this _MUST_ be the _ONLY_ place in daru where the
     # @dtype variable is set and the underlying data type of vector changed.
-    def cast_vector_to dtype, source=nil, nm_dtype=nil
+    def cast_vector_to dtype, source=nil, nm_dtype=nil # rubocop:disable Style/CyclomaticComplexity
       source = @data.to_a if source.nil?
 
       new_vector =
         case dtype
         when :array   then Daru::Accessors::ArrayWrapper.new(source, self)
-        when :nmatrix then Daru::Accessors::NMatrixWrapper.new(source, self, nm_dtype)
+        when :nmatrix then
+          # To avoid arrays with nils throwing TypeError for :int32 default nm_dtype
+          nm_dtype = :object if nm_dtype.nil? && source.any?(&:nil?)
```

Comment:
The downside is if I do it there I'd either have to remove the `nm_dtype :int32` default in the method signature (which would mean doing the `any?(&:nil?)` check every time the method is called without an explicit `nm_dtype` (not good). Or I would have to check for `nm_dtype == :int32` which might produce unexpected results for someone that's deliberately passed that and might reasonably expect the method to fail.